src/libs/actions/Policy.ts

```diff
@@ -563,6 +565,100 @@ function setWorkspaceApprovalMode(policyID: string, approver: string, approvalMo
     API.write(WRITE_COMMANDS.SET_WORKSPACE_APPROVAL_MODE, params, {optimisticData, failureData, successData});
 }
 
+function setWorkspacePayer(policyID: string, reimburserEmail: string) {
+    const policy = ReportUtils.getPolicy(policyID);
+
+    const currentPolicyReimburserEmail = policy.reimburserEmail;
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                reimburserEmail,
+                errorFields: {reimburserEmail: null},
+                pendingFields: {reimburserEmail: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                reimburserEmail: currentPolicyReimburserEmail,
+                errorFields: {reimburserEmail: ErrorUtils.getMicroSecondOnyxError('workflowsPayerPage.genericErrorMessage')},
+                pendingFields: {reimburserEmail: null},
+            },
+        },
+    ];
```

Comment:
NAB but we usually define `optimisticData`, `successData` and `failureData` in that order, so it'd be nice to move this below `successData`